src/shared/Combobox.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from "react";
+import { ActionMeta, ValueType } from "react-select";
+import CreatableSelect from "react-select/creatable";
+
+export type ComboboxItem = {
+	value: any;
+	label: string;
+};
+
+enum OnChangeEnum {
+  selectOption = "select-option",
+  deselectOption = "deselect-option",
+  removeValue = "remove-value",
+  popValue = "pop-value",
+  setValue = "set-value",
+  clear = "clear",
+  createOption = "create-option"
+}
+
+type ComboboxProps = {
+  itemsSource: any[]; 
+  selectedItem: any;
+  onSelectValue: (selectedItem: any) => void;
```

Comment:
I understand your mindflow and it's fine when you Combobox could handle multiple types of dataSource but we definitely should know what exactly data type we want to handle in some particular place.
In this case, I'd rather use some generic Props instead just too common _any_

Something like that

```
type Option<T> = {
  value: T;
  label: string;
};

type Props<T> = {
  options: Option<T>[];
  selectedItem: T; // our selected item will be exactly that type what we expect to have in this place
  onChange: (value: T) => void; // we expect to receive the same type what we handles by this Combobox.
};

function Combobox<T extends OptionValue>(props: Props<T>) { ..... }

Combobox Call
<Combobox<CustomType or string or whatever>
      options={.....}
      selectedItem={target}
      onChange={setTarget}
    />
```

Reply:
This is really cool.   I'll make a card for you to add this after I check in this PR.